tests/js/common/schema-holder-tests.js

```diff
@@ -0,0 +1,247 @@
+/* eslint-env browser */
+/* globals require */
+var fluid  = fluid || {};
+var jqUnit = jqUnit || {};
+
+(function (fluid, jqUnit) {
+    "use strict";
+    if (!fluid.identity) {
+        fluid  = require("infusion");
+        jqUnit = require("node-jqunit");
+        require("../../../src/js/common/schemaHolder");
+    }
+    var gpii = fluid.registerNamespace("gpii");
+
+    // A schema holder with no children.
+    fluid.defaults("gpii.tests.schema.schemaHolder", {
+        gradeNames: ["gpii.schema.schemaHolder"],
+        schema: {
+            type: "object",
+            properties: {
+                childProperty: {type: "string"}
+            }
+        }
+    });
+
+    // This transform is a crude demonstration of what we might do with the LSR, where we need to store multiple
+    // settings in a generated supportedSettings block.
+    fluid.defaults("gpii.tests.schema.schemaHolder.transforms.mergeChild", {
+        gradeNames: ["fluid.transformFunction"]
+    });
+
+    fluid.registerNamespace("gpii.tests.schema.schemaHolder.transforms");
+    gpii.tests.schema.schemaHolder.transforms.mergeChild = function (transformSpec, transformer) {
+        if (!transformSpec.toMerge) {
+            fluid.fail("mergeChild transform requires a `toMerge` inputPath.", transformSpec);
+        }
+        var toMerge = transformer.expand(transformSpec.toMerge);
+        // TODO: pass both components rather than just their schemas, so that we can use some aspect of the component
+        //  rather than the "title" property of the schema.
+        var outputPath = fluid.model.composePaths(transformer.outputPrefix, toMerge.title);
+        transformer.applier.change(outputPath, toMerge);
```

Comment:
This is just too much - we can't have a transform rule which reaches out in this side-effectful way to the transformer's machinery

Reply:
We should review in today's meeting, this was just a first pass.  I need some way of constructing a map of material from child components based potentially on some property of theirs.    I need an approach like this to generate `supportedSettings`, `capabilitiesTransforms`, and `inverseCapabilitiesTransforms` blocks within a "solution component" (or perhaps a "launch handler component") based on whatever child "setttings components" are found.